### PR TITLE
fix: validation and documentation for unsupported Windows platform

### LINKED ISSUE #1611: Ensure Windows support
Incoming from the forum:
 - https://forum.fuel.network/t/3938

Potential causes inside the `install` script for `forc` and `fuel-core`:
 1. Use of `tar` _([here](https://github.com/FuelLabs/fuels-ts/blob/df3222d5cd98fd87faee7e9197d17e7a0d274da5/packages/forc/lib/install.js#L56) and [here](https://github.com/FuelLabs/fuels-ts/blob/df3222d5cd98fd87faee7e9197d17e7a0d274da5/packages/fuel-core/lib/install.js#L66))_
    - Consider using: https://www.npmjs.com/package/tar

 1. Use of `rm -rf` _([here](https://github.com/FuelLabs/fuels-ts/blob/df3222d5cd98fd87faee7e9197d17e7a0d274da5/packages/fuel-core/lib/install.js#L55) and [here](https://github.com/FuelLabs/fuels-ts/blob/df3222d5cd98fd87faee7e9197d17e7a0d274da5/packages/fuel-core/lib/install.js#L72))_
    - Consider using `rmSync` _([example](https://github.com/FuelLabs/fuels-ts/blob/df3222d5cd98fd87faee7e9197d17e7a0d274da5/packages/forc/lib/install.js#L59))_

> A full checkup is necessary.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Added a notice in the docs and readme around Windows support
- Refactored the installation scripts to use functions from the Node.js `fs` module instead of hard-coded UNIX commands like `tar`
- Fixed the installation script erroring out on Windows without logging the correct error

> Please excuse the dirty commit history. Debugging this was not pretty lol

Closes #1611

### CODE REVIEW

File: packages/create-fuels/package.json [hunk 1]
```diff
@@ -22,12 +22,10 @@
     "chalk": "4",
     "commander": "^9.4.1",
     "fuels": "workspace:*",
-    "prompts": "^2.4.2",
-    "tar": "^6.2.0"
```
Comment (Dhaiwat10) [inline packages/create-fuels/package.json:26]: Removed this unused dependency that I came across while scanning for the usage of the `tar` command

**Comment by arboleya** [comment]
What do you think of using this title instead?
 - `fix: validation and documentation for unsupported Windows platform`

**Review by arboleya — COMMENTED** [review]
What do you think of using this title instead?
 - `fix: validation and documentation for unsupported Windows platform`

**Review by danielbate — APPROVED** [review]
LGTM! A shame we can't support windows though.

**Comment by danielbate** [comment]
Should this be targeting `rc/salamander`?

**Comment by Dhaiwat10** [comment]
@danielbate we wanted to release this fix directly to master since it could be critical, rather than waiting for the RC

**Review by arboleya — APPROVED** [review]

**Pull request merged**